File: golang/73-set-matrix-zeroes.go
package golang

func setZeroes(matrix [][]int)  {
    n, m := len(matrix), len(matrix[0])
    zeroRows := make([]bool, n)
    zeroCols := make([]bool, m)
    
    for x := 0; x < n; x++ {
        for y := 0; y < m; y++ {
            if matrix[x][y] == 0 {
                zeroRows[x] = true
                zeroCols[y] = true
            }
        }
    }
    
    for i := 0; i < n; i++ {
        if zeroRows[i] {
            for j := 0; j < m; j++ {
                matrix[i][j] = 0
            }
        }
    }
    
    for i := 0; i < m; i++ {
        if zeroCols[i] {
            for j:= 0; j < n; j++ {
                matrix[j][i] = 0
            }
        }
    }
}

File: typescript/73-set-matrix-zeroes.ts
/**
 Do not return anything, modify matrix in-place instead.
 */
 function setZeroes(matrix: number[][]): void {
    const zeroRows = (new Array(matrix.length)).fill(false)
    const zeroCols = (new Array(matrix[0].length)).fill(false)
    for (let x = 0; x < matrix.length; x++) {
        for (let y = 0; y < matrix[x].length; y++) {
            if (matrix[x][y] === 0) {
                zeroRows[x] = true
                zeroCols[y] = true
            }
        }
    }        
    for (let i = 0; i < zeroRows.length; i++) {
        if (zeroRows[i] === true) {
            for (let j = 0; j < matrix[i].length; j++) {
                matrix[i][j] = 0
            }
        }
    }
    for (let i = 0; i < zeroCols.length; i++) {
        if (zeroCols[i] === true) {
            for (let j = 0; j < matrix.length; j++) {
                matrix[j][i] = 0
            }
        }
    }
};